substratevm/src/org.graalvm.polyglot.nativeapi/src/org/graalvm/polyglot/nativeapi/PolyglotNativeAPI.java

```diff
@@ -1430,16 +1431,16 @@ public static PolyglotStatus poly_language_get_id(PolyglotIsolateThread thread,
                     " @since 1.0",
     })
     public static PolyglotStatus poly_get_last_error_info(PolyglotIsolateThread thread, @CConst PolyglotExtendedErrorInfoPointer result) {
-        ErrorInfoHolder errorInfoHolder = errorInfo.get();
-        if (errorInfoHolder != null) {
-            result.write(errorInfoHolder.info);
+        ErrorStateHolder holder = errorHolder.get();
+        if (holder != null && holder.info.isNonNull()) {
+            result.write(holder.info);
             return poly_ok;
         } else {
             return poly_generic_failure;
```

Comment:
Since we return null if there is no exception, should we do the same here? What does `napi` do?

Reply:
They do not check if an error had occurred, and return napi_ok unless a NULL environment had been passed. I agree, we should rewrite this one to behave in the same way too, doing it now.